File: app/src/main/java/fr/xgouchet/chronorg/ui/presenters/EventListPresenter.java
package fr.xgouchet.chronorg.ui.presenters;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.deezer.android.counsel.annotations.Trace;

import fr.xgouchet.chronorg.data.models.Event;
import fr.xgouchet.chronorg.data.models.Project;
import fr.xgouchet.chronorg.data.repositories.EventRepository;
import fr.xgouchet.chronorg.ui.contracts.presenters.BaseListPresenter;
import rx.Observable;

/**
 * @author Xavier Gouchet
 */
@Trace
public class EventListPresenter extends BaseListPresenter<Event> {

    @NonNull private final EventRepository eventRepository;
    @Nullable private Project project;

    public EventListPresenter(@NonNull EventRepository eventRepository) {
        this.eventRepository = eventRepository;
    }

    public void setProject(@NonNull Project project) {
        this.project = project;
    }

    @Override protected Observable<Event> getItemsObservable() {
        return eventRepository.getEventsInProject(project.getId());
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/feature/events/EventListPresenter.kt
package fr.xgouchet.khronorg.feature.events

import fr.xgouchet.khronorg.feature.projects.Project
import fr.xgouchet.khronorg.feature.events.Event
import fr.xgouchet.khronorg.commons.query.QueryBuilder
import fr.xgouchet.khronorg.commons.repositories.BaseRepository
import fr.xgouchet.khronorg.provider.KhronorgSchema.Companion.COL_NAME
import fr.xgouchet.khronorg.provider.KhronorgSchema.Companion.COL_PROJECT_ID
import fr.xgouchet.khronorg.feature.events.EventNavigator
import fr.xgouchet.khronorg.ui.presenters.BaseListPresenter
import io.reactivex.Observable

/**
 * @author Xavier F. Gouchet
 */
class EventListPresenter(val repository: BaseRepository<Event>, val project: Project, navigator: EventNavigator)
    : BaseListPresenter<Event>(navigator) {


    override fun getItemsObservable(): Observable<Event> {
        val alteration = QueryBuilder.where {
            equals(COL_PROJECT_ID, project.id.toString())
            orderAsc(COL_NAME)
        }

        return repository.getWhere(alteration)
    }

}